scss/_patterns_subnav.scss

```diff
@@ -33,13 +33,21 @@
     }
   }
 
-  // XXX: Fix required because of greedy rule in p-navigation
-  // Can be updated with resolution of:
-  // https://github.com/canonical-web-and-design/vanilla-framework/issues/2168
-  .p-subnav > a {
+  %subnav-link {
     padding-right: 2 * $sph-inner--small + map-get($icon-sizes, default); // icon padded with the default padding-right of selects, inputs etc.
   }
 
+  .p-subnav > .p-navigation__link {
```

Comment:
Seems strange to cross use two patterns here?

Reply:
It's been like this before. I don't want to change it in this PR. I opened a separate issue to move p-subnav into navigation #2857.